modules/imgproc/src/filter.simd.hpp

```diff
@@ -1117,6 +1117,81 @@ struct SymmColumnVec_32s8u
     Mat kernel;
 };
 
+struct SymmColumnVec_32f8u
+{
+    SymmColumnVec_32f8u() { symmetryType = 0; delta = 0; }
+    SymmColumnVec_32f8u(const Mat& _kernel, int _symmetryType, int, double _delta)
+    {
+        symmetryType = _symmetryType;
+        kernel = _kernel;
+        delta = (float)_delta;
+        CV_Assert( (symmetryType & (KERNEL_SYMMETRICAL | KERNEL_ASYMMETRICAL)) != 0 );
+    }
+
+    int operator()(const uchar** _src, uchar* _dst, int width) const
+    {
+        CV_INSTRUMENT_REGION();
+
+        int _ksize = kernel.rows + kernel.cols - 1;
+        if( _ksize == 1 ) return 0;
+        const int ksize2 = _ksize / 2;
+        const float* ky = kernel.ptr<float>() + ksize2;
+        int i = 0, k;
+        bool symmetrical = (symmetryType & KERNEL_SYMMETRICAL) != 0;
+        const float** src = (const float**)_src;
+
+        if( symmetrical )
+        {
+            for( ; i <= width - v_uint8::nlanes; i += v_uint8::nlanes )
+            {
+                v_float32 v_ky0 = vx_setall_f32(ky[0]);
+                v_float32 v32_delta = vx_setall_f32(delta);
+                const float* S = src[0] + i;
+                v_float32 s0 = v_muladd(v_ky0, vx_load(S), v32_delta);
+                v_float32 s1 = v_muladd(v_ky0, vx_load(S + v_float32::nlanes), v32_delta);
+                v_float32 s2 = v_muladd(v_ky0, vx_load(S + 2*v_float32::nlanes), v32_delta);
+                v_float32 s3 = v_muladd(v_ky0, vx_load(S + 3*v_float32::nlanes), v32_delta);
+                for( k = 1; k <= ksize2; k++ )
+                {
+                    v_float32 v_kyk = vx_setall_f32(ky[k]);
+                    const float* S0 = src[k] + i;
+                    const float* S1 = src[-k] + i;
+                    s0 = v_muladd(v_kyk, vx_load(S0) + vx_load(S1), s0);
+                    s1 = v_muladd(v_kyk, vx_load(S0 + v_float32::nlanes) + vx_load(S1 + v_float32::nlanes), s1);
+                    s2 = v_muladd(v_kyk, vx_load(S0 + 2*v_float32::nlanes) + vx_load(S1 + 2*v_float32::nlanes), s2);
+                    s3 = v_muladd(v_kyk, vx_load(S0 + 3*v_float32::nlanes) + vx_load(S1 + 3*v_float32::nlanes), s3);
+                }
+                v_store(_dst + i, v_pack_u(v_pack(v_round(s0), v_round(s1)), v_pack(v_round(s2), v_round(s3))));
```

Comment:
IMHO, saturation is not properly handled here.

Looks like there is no accuracy tests for this code path in imgproc's tests.

Reply:
> IMHO, saturation is not properly handled here.

Could I get some clarification on this please? My understanding is that `v_pack` uses saturation on all but 64-bit int inputs so there shouldn't be a problem here? Any suggestions on how to better write this?

> Looks like there is no accuracy tests for this code path in imgproc's tests.

Are you referring to the case when the kernel is symmetrical? If yes, do you know which tests in `opencv_test_imgproc` can be modified to test this code path?

Reply:
If there are no tests for the symmetrical case, either in the `imgproc` module itself or in any of the other modules that depends on `imgproc`, would it make sense to replace the SIMD code in the `if (symmetrical )` case with something along the lines of `CV_Assert("Symmetrical case is not implemented?")`.

If the assert is never raised in any of the test or perf executables, it means that it is probably OK for now to postpone the implementation of the symmetrical case as it does not seem to be used?

Alternatively, instead of an assert, maybe a warning followed by `return i;`? That would make the code functionally correct, it would just rely on the original scalar version.

Reply:
Gentle ping @alalek. What are your thoughts on this?